sklearn/preprocessing/_data.py

```diff
@@ -70,7 +73,7 @@ def _handle_zeros_in_scale(scale, copy=True):
         if scale == .0:
             scale = 1.
         return scale
-    elif isinstance(scale, np.ndarray):
+    elif isinstance(scale, (np.ndarray, cp.ndarray)):
```

Comment:
The goal of this exercise is to rewrite the code to benefit from ducktyping while not introducing a new dependency on cupy or any other NEP-18 enabled library.

So here we could just assume that if scale is not scalar, it's a numpy like array. So we just replace this `elif` by and `else`.

Reply:
Yes, of course. I think though that the function is suppose to return `None` when the input is neither a scalar or a numpy-like array. In this case, it's maybe better to use `if hasattr(X, "__array_function__"):` . What do you think?

Reply:
+1